js/src/logger.ts

```diff
@@ -2301,9 +2350,12 @@ export class Dataset extends ObjectFetcher<DatasetRecord> {
 
   constructor(
     lazyMetadata: LazyValue<ProjectDatasetMetadata>,
-    pinnedVersion?: string
+    pinnedVersion?: string,
+    outputInsteadOfExpected?: boolean,
   ) {
-    super("dataset", pinnedVersion);
+    const outputInsteadOfExpectedBool = outputInsteadOfExpected || outputInsteadOfExpected === undefined;
```

Comment:
Alternatively, you can use the "null-ish" coalescing operator `outputInsteadOfExpected ?? true`, which will only use the right hand operand if the left is null or undefined.